build-system/server.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Creates an http server to handle static
+ * files and list directories for use with the gulp live server
+ */
+var app = require('connect')();
+var fs = require('fs');
+var path = require('path');
+var clr = require('connect-livereload');
+var serveStatic = require('serve-static');
+var url = require('url');
+
+var args = Array.prototype.slice.call(process.argv, 2, 4);
+var paths = args[0];
+var port = args[1];
+
+app.use(clr());
+
+// Serve static files that exist
+paths.split(",").forEach(function(pth){
+    app.use(serveStatic(path.join(process.cwd(), pth)));
+});
+
+app.use(function(req, res) {
+  var pathname = url.parse(req.url).pathname
+  var filename = path.join(process.cwd(), pathname);
+  path.exists(filename, function (exists) {
+
+    // files that don't actually exist
+    if(!exists) {
+      res.writeHead(404, { 'Content-Type': 'text/plain' });
+      res.write('404: Page not Found');
+      res.end();
+      return;
+    }
+
+    // handle directory listings
+    fs.readdir(filename, function (err, files) {
+      res.writeHead(200);
+      res.write('<html><head><title>' + pathname + '</title>');
```

Comment:
Sure thing. I'm going to peek into that serve-index some more and see what it can do; Looks like it could give a prettier looking list page if nothing else.